eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -1,76 +1,40 @@
-# cloud instance. Normally we would use template parameters instead of variables
-# but matrix variables are not available during template expansion so any
-# benefits of parameters are lost.
-
 parameters:
   ServiceDirectory: not-set
   ArmTemplateParameters: '@{}'
   DeleteAfterHours: 24
   Location: ''
+  SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+
+# SubscriptionConfiguration will be splat into the parameters of the test
+# resources script. It should be JSON in the form:
+# {
+#   "SubscriptionId": "<subscription id>",
+#   "TenantId": "<tenant id>",
+#   "TestApplicationId": "<test app id>",
+#   "TestApplicationSecret": "<test app secret>",
+#   "ProvisionerApplicationId": "<provisoner app id>",
+#   "ProvisoinerApplicationSecret": "<provisoner app secert>",
+#   "Environment": "AzureCloud | AzureGov | AzureChina | <other environment>"
+# }
 
 steps:
   # New-TestResources command requires Az module
   - pwsh: Install-Module -Name Az -Scope CurrentUser -AllowClobber -Force -Verbose
     displayName: Install Azure PowerShell module
 
-  - pwsh: >
-      eng/common/TestResources/New-TestResources.ps1
-      -BaseName 'Generated'
-      -ServiceDirectory '${{ parameters.ServiceDirectory }}'
-      -TenantId '$(aad-azure-sdk-test-tenant-id)'
-      -SubscriptionId '$(azure-subscription-id)'
-      -TestApplicationId '$(aad-azure-sdk-test-client-id)'
-      -TestApplicationSecret '$(aad-azure-sdk-test-client-secret)'
-      -ProvisionerApplicationId '$(aad-azure-sdk-test-client-id)'
-      -ProvisionerApplicationSecret '$(aad-azure-sdk-test-client-secret)'
-      -AdditionalParameters ${{ parameters.ArmTemplateParameters }}
-      -DeleteAfterHours ${{ parameters.DeleteAfterHours }}
-      -Location '${{ parameters.Location }}'
-      -Environment 'AzureCloud'
-      -CI
-      -Force
-      -Verbose
-    displayName: Deploy test resources (AzureCloud)
-    condition: and(succeeded(), eq(variables['CloudType'], 'AzureCloud'))
-
-  - pwsh: >
-      eng/common/TestResources/New-TestResources.ps1
-      -BaseName 'Generated'
-      -ServiceDirectory '${{ parameters.ServiceDirectory }}'
-      -TenantId '$(aad-azure-sdk-test-tenant-id-gov)'
-      -SubscriptionId '$(azure-subscription-id-gov)'
-      -TestApplicationId '$(aad-azure-sdk-test-client-id-gov)'
-      -TestApplicationSecret '$(aad-azure-sdk-test-client-secret-gov)'
-      -ProvisionerApplicationId '$(aad-azure-sdk-test-client-id-gov)'
-      -ProvisionerApplicationSecret '$(aad-azure-sdk-test-client-secret-gov)'
-      -AdditionalParameters ${{ parameters.ArmTemplateParameters }}
-      -DeleteAfterHours ${{ parameters.DeleteAfterHours }}
-      -Location '${{ parameters.Location }}'
-      -Environment 'AzureUSGovernment'
-      -CI
-      -Force
-      -Verbose
-    displayName: Deploy test resources (AzureUSGovernment)
-    condition: and(succeeded(), eq(variables['CloudType'], 'AzureUSGovernment'))
+  - pwsh: |
+      $subscriptionConfiguration = @"
```

Comment:
Why use a string literal? In fact, because quotes will be in there, use single quotes like so and the line would read easier as a one-liner:
```yaml
- pwsh: $subscriptionConfiguration = '${{ parameters.SubscriptionConfiguration}}' | ConvertFrom-Json -AsHashtable
```

Reply:
Client secret strings can have single quotes in them and this approach will cause PowerShell to treat the single quote in the secret as the end of the string.